crates/mod-host/src/asset_archive.rs

```diff
@@ -0,0 +1,70 @@
+use std::cell::OnceCell;
+use std::sync::{Arc, LazyLock, Mutex};
+use std::fs::File;
+use std::io::Write;
+
+use retour::Function;
+use thiserror::Error;
+use me3_mod_protocol::package::Package;
+use me3_mod_host_assets::mapping::ArchiveOverrideMapping;
+use me3_mod_host_assets::ffi::{get_dlwstring_contents, set_dlwstring_contents, DLWString};
+
+use crate::{
+    detour::{install_detour, Detour, DetourError},
+    host::ModHost,
+};
+
+/// This is a heavily obfuscated std::basic_string replace-esque. It's only 
+/// used when the game wants to expand a path as part of the archive lookup.
+pub type ExpandArchivePathFn = extern "C" fn(*mut DLWString, usize, usize, usize, usize, usize);
+
+pub fn attach(host: &mut ModHost, mapping: ArchiveOverrideMapping) -> Result<(), DetourError> {
+    let hook_instance: Arc<OnceCell<Arc<Detour<ExpandArchivePathFn>>>> = Default::default();
+
+    let hook = {
+        let hook_instance = hook_instance.clone();
+
+        host.hook(get_hook_location())
+            .with_closure(move |path, p2, p3, p4, p5, p6| {
+                // Have the game expand the path for us.
+                hook_instance.get().unwrap().trampoline()(path, p2, p3, p4, p5, p6);
+
+                let resource_path_string = get_dlwstring_contents(unsafe {
+                    path.as_mut().unwrap()
+                });
+
+                writeln!(
+                    &mut LOG.lock().unwrap(),
+                    "Asset request: {}",
+                    unsafe { get_dlwstring_contents(path.as_ref().unwrap()) },
+                );
+
+                // Match the expanded path against the known overrides.
+                if let Some(mapped_override) = mapping.get_override(&resource_path_string) {
+                    // Replace the string with a canonical path to the asset if
+                    // we did find an override. This will cause the game to 
+                    // pull the files bytes from the file system instead of the
+                    // BDTs.
+                    set_dlwstring_contents(unsafe { path.as_ref().unwrap() }, mapped_override);
+
+                    writeln!(
```

Comment:
Should use `tracing` crate log functions.

Reply:
Fixed in [57c7bf](https://github.com/garyttierney/me3/pull/22/commits/157c7bfffffbf886bb184d7cc915aff16e1daf14)